File: kotlin/abc081_b.kt
// ABC081B - Shift only

fun main() {
  val N = readLine()!!.toInt()
  val A = readLine()!!.split(" ").map(String::toInt).toIntArray()
  var cnt = 0
  while (A.all { it % 2 == 0 }) {
    repeat(N) { A[it] /= 2 }
    cnt++
  }
  println(cnt)
}

File: javascript/abc081_b.js
// ABC081B - Shift only

const input = require("fs").readFileSync("/dev/stdin", "utf8").split("\n");
const N =Number(input[0]);
const A = input[1].split(" ").map(Number);

let cnt = 0
while (A.every(a => a % 2 === 0)) {
  [...Array(N)].map((_, i) => A[i] /= 2);
  cnt++;
}

console.log(cnt);
